File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.alibaba.dubbo.config.annotation.Service;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * 品牌controller
 */
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    /**
     * 返回全部列表
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    /**
     * 分页查询
     *
     * @param num
     * @param size
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int num, int size) {
        return brandService.findPage(num, size);
    }

    /**
     * 条件查询+分页
     *
     * @param num
     * @param size
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(int num, int size, @RequestBody TbBrand tbBrand) {
        return brandService.search(num, size, tbBrand);
    }

    /**
     * 增加
     *
     * @param tbBrand
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand) {

        try {
            brandService.add(tbBrand);
            return new Result(true, "成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "失败");
        }
    }

    /**
     * 修改
     *
     * @param tbBrand
     * @return
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand) {
        try {
            brandService.update(tbBrand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    /**
     * 获取实体
     *
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(long id) {
        TbBrand one = brandService.findOne(id);
        return one;
    }

    /**
     * 批量删除
     *
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(long[] ids) {
        try {
            brandService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }

    @RequestMapping("/findOptionList")
    public List<Map> findOptionList(){
        return brandService.findOptionList();
    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function ($scope,$controller,brandService) {

    $controller('baseController',{$scope:$scope});


    //读取列表数据绑定到表单中
    // $scope.findAll=function () {
    //     brandService.findAll().success(
    //         function (response) {
    //             $scope.list=response;
    //         }
    //     );
    // };


    //分页查询
    $scope.findPage=function (num,size) {
        brandService.findPage(num,size).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        );
    };

    //保存
    $scope.save=function () {
        var object;
        if($scope.entity.id != null){
            //执行修改方法
            object =  brandService.update($scope.entity);
        }else{
            //增加
            object = brandService.add($scope.entity);
        }
        object.success(
            function (response) {
                if(response.success){
                    //重新查询 加载
                    $scope.reloadList();
                }else {
                    alert(response.message)
                }
            }
        );
    };
    //获取实体
    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    };



    //批量删除
    $scope.delete=function () {
        brandService.delete($scope.selectIds).success(
            function (response) {
                if(response.success){
                    $scope.reloadList();//删除成功 刷新列表
                    $scope.selectIds=[];
                }
            }
        );
    };

    //分页多条件查询
    $scope.searchentity={};
    $scope.search=function (num,size) {
        brandService.search(num,size,$scope.searchentity).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        );
    }

});
